tooling/SecretManager.VisualStudio/SecretManagerFactory.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Extensions.SecretManager.VisualStudio
+{
+    internal class SecretManagerFactory 
+    {
+        private readonly Lazy<ProjectLocalSecretsManager> _secretManager;
+        private readonly ConfiguredProject _project;
+
+        [ImportingConstructor]
+        public SecretManagerFactory(ConfiguredProject project)
```

Comment:
Is configured project the right scope for this? I would expect it to be unconfigured project.

ConfiguredProject = 1 instance per project file X debug or release X bitness
UnconfiguredProject = 1 instance per projectfile

Reply:
Yes. Although this is uncommon, the following would be a legitimate way to use UserSecrets.
```xml
<PropertyGroup>
  <UserSecretsId Condition="'$(Configuration)' == 'Debug'">debug-secrets-id</UserSecretsId>
  <UserSecretsId Condition="'$(Configuration)' == 'Release'">release-secrets-id</UserSecretsId>
</PropertyGroup>
```

Reply:
Only if we decide to support it.